### ISSUE TITLE
Does `nvs` not support using `nix run` / `nix shell`?

### ISSUE THREAD

**Issue by kanielrkirby** [jeff-hykin/nix_version_search_cli#8]
I find this to be strange behavior from a `nix` focused project. Is there a reason this might not be supported / currently work?

```bash
❯ nix run github:jeff-hykin/nix_version_search_cli
Could not start dynamically linked executable: /nix/store/lnpwfdzjmrjwcijjcrlyxrg7zy8iyg4c-nvs/bin/nvs
NixOS cannot run dynamically linked executables intended for generic
linux environments out of the box. For more information, see:
https://nix.dev/permalink/stub-ld

❯ nix shell github:jeff-hykin/nix_version_search_cli

❯ nvs
Could not start dynamically linked executable: /nix/store/lnpwfdzjmrjwcijjcrlyxrg7zy8iyg4c-nvs/bin/nvs
NixOS cannot run dynamically linked executables intended for generic
linux environments out of the box. For more information, see:
https://nix.dev/permalink/stub-ld
```

**Comment by jeff-hykin** [comment]
Thanks for letting me know this doesn't work.

> Is there a reason this might not be supported / currently work?

Yes. Right now, inside the nix derivation, the binary is compiled using Deno. And deno definitely doesnt have a static linking option. So the final executable links to a bunch of standard Linux shared object files (probably). And thats what NixOS is complaining about. 

Theres probably a way to patch it with patchelf but I'm not sure how much elbow greese thats going to need. I'll need to find equivalent .so files inside the nix store, add those as runtime dependencies, and patch the binary so it dynamically links to them instead of standard Linux ones.

Usually this would be handled by autoPatchElf in mkDerivation, but mkDerivation breaks the deno executable, so I have to call derivation manually, which means I also need to manually do the Patchelf steps.

**Comment by kanielrkirby** [comment]
Ah, I see. I was going to ask if `deno2nix` would be a solution, but it seems to be unmaintained unfortunately, with no definitive fork.

**Comment by kanielrkirby** [comment]
There's a few (very few) `deno` packages in [`NixOS/nixpkgs`](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs%20deno&type=code) that could potentially be some inspiration as well, if you end up wanting to solve this. Unfortunately, systems with Deno are a bit outside my expertise, or I'd happily mess with it a bit / make a PR.

**Comment by jeff-hykin** [comment]
I don't think this is a deno specific problem, I just need to patch the dynamic links. I appreciate the effort to help though!

I found theres 4 dynamic links, all go to the nix store, but 3 are not hardcoded to the nix store so I'm assuming those are the ones I need to fix with patchelf

**Comment by senorsmile** [comment]
I encountered this project, tried to run the same thing, and got the same errors.  

I get the same error if I try to nix profile install per the README.  

I git cloned, tried to enter a develop env, and got 

```
error: builder for '/nix/store/jydvy0jlm3sy8c50vz3ip349xjxcc5s8-nvs-env.drv' failed to produce output path for output 'out' at '/nix/store/jydvy0jlm3sy8c50vz3ip349xjxcc5s8-nvs-env.drv.chroot/root/nix/store/298aljzyxy9s3hisj57lx0rz16alhiyy-nvs-env'
```

I tried current master, and checked out several versions.

**Comment by jeff-hykin** [comment]
Okay I think this is fixed now, let me know if its still failing for you guys:

```sh
nix-env -i -f https://github.com/jeff-hykin/nix_version_search_cli/archive/50a3fef5c9826d1e08b360b7255808e53165e9b2.tar.gz
# or, if you have flakes:
nix profile install 'https://github.com/jeff-hykin/nix_version_search_cli/archive/50a3fef5c9826d1e08b360b7255808e53165e9b2.tar.gz#nvs'
```

**Comment by reisholmes** [comment]
> Okay I think this is fixed now, let me know if its still failing for you guys:
> 
> nix-env -i -f https://github.com/jeff-hykin/nix_version_search_cli/archive/50a3fef5c9826d1e08b360b7255808e53165e9b2.tar.gz
>  or, if you have flakes:
> nix profile install 'https://github.com/jeff-hykin/nix_version_search_cli/archive/50a3fef5c9826d1e08b360b7255808e53165e9b2.tar.gz#nvs'

Just tested this with `nix shell https://github.com/jeff-hykin/nix_version_search_cliarchive/50a3fef5c9826d1e08b360b7255808e53165e9b2.tar.gz' and it works exactly as expected with the functionality I tried 👍🏾

**Comment by jeff-hykin** [comment]
Thank you for testing!
